packages/flutter/lib/src/services/text_input.dart

```diff
@@ -609,6 +611,47 @@ class TextInputConfiguration {
   /// {@endtemplate}
   final bool enableIMEPersonalizedLearning;
 
+  /// Used when a user inserts image-based content through the device keyboard
+  /// on Android only.
+  ///
+  /// The passed list of strings will determine which MIME types are allowed to
+  /// be inserted via the device keyboard
+  ///
+  /// This example shows how to limit your keyboard commits to specific file types
+  /// `TextField`.
+  ///
```

Comment:
Nit: Maybe somewhere in here we should mention the behavior if it's left empty, like what you explained in your comment.